scripts/pureshear_ALE.jl

```diff
@@ -0,0 +1,141 @@
+using Statistics, LinearAlgebra, Printf, Base.Threads, CairoMakie
+const year     = 365*3600*24
+const USE_GPU  = false
+const GPU_ID   = 0
+const USE_MPI  = false
+
+using JustPIC, JustPIC._2D
+const backend = JustPIC.CPUBackend 
+
+ALE    = false
+
+using ParallelStencil
+using ParallelStencil.FiniteDifferences2D
+@static if USE_GPU
+    @init_parallel_stencil(CUDA, Float64, 3)
+    CUDA.device!(GPU_ID) # select GPU
+else
+    @init_parallel_stencil(Threads, Float64, 3)
+end
+
+@parallel_indices (I...) function InitialFieldsParticles!( phases, px, py, index)
+    @inbounds for ip in cellaxes(phases)
+        # quick escape
+        @cell(index[ip, I...]) == 0 && continue
+        x = @cell px[ip, I...]
+        y = @cell py[ip, I...]
+        if x<y
+            @cell phases[ip, I...] = 1.0
+        else
+            @cell phases[ip, I...] = 2.0
+        end
```

Comment:
```suggestion
        @index(index[ip, I...]) == 0 && continue
        x = @index px[ip, I...]
        y = @index py[ip, I...]
        if x<y
            @index phases[ip, I...] = 1.0
        else
            @index phases[ip, I...] = 2.0
        end
```
We should use the new macros now.